cdap-app-templates/cdap-etl/cdap-etl-api/src/main/java/co/cask/cdap/etl/api/StageConfigurer.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.etl.api;
+
+import co.cask.cdap.api.data.schema.Schema;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * This stores the input schemas that is passed to this stage from other stages in the pipeline and
+ * the output schemas that could be sent to the next stages from this stage.
+ */
+public class StageConfigurer {
+  private List<Schema> outputSchemas;
+  private List<Schema> inputSchemas;
+
+  public StageConfigurer() {
+    this.outputSchemas = new ArrayList<>();
+    this.inputSchemas = new ArrayList<>();
+  }
+
+  /**
+   * list of possible output schema from this stage
+   * @return
+   */
+  public List<Schema> getOutputSchemas() {
+    return outputSchemas;
+  }
+
+  /**
+   * list of possible input schemas sent to this stage
+   * @return
+   */
+  public List<Schema> getInputSchemas() {
+    return inputSchemas;
+  }
+
+  /**
+   * add the passed output schema to the list of output schema sent from this stage.
+   * @param outputSchema
+   */
+  public void addOutputSchema(Schema outputSchema) {
+    this.outputSchemas.add(outputSchema);
+  }
+
+  /**
+   * add the passed list to the list of output schema sent from this stage.
+   * @param outputSchema
+   */
+  public void addOutputSchemas(List<Schema> outputSchema) {
+    this.outputSchemas.addAll(outputSchema);
+  }
+
+  /**
+   * list of input schemas that can be sent to this stage.
+   * @param inputSchema
+   */
+  public void addInputSchemas(List<Schema> inputSchema) {
```

Comment:
setInputSchema should not be in the interface in api since it should only be visible to the application but not to the plugins.  This method should exist in the implementation but not here.

Also need to add a check to see if the input schema was already set and if it was set, error if the schemas are different.